vllm/model_executor/layers/quantization/awq.py

```diff
@@ -153,7 +153,16 @@ def apply_weights(self,
         pack_factor = self.quant_config.pack_factor
         out_shape = (x.shape[:-1] + (qweight.shape[-1] * pack_factor, ))
         reshaped_x = x.reshape(-1, x.shape[-1])
-        out = ops.awq_gemm(reshaped_x, qweight, scales, qzeros, pack_factor)
+
+        # batch_size*seq_len >= threshold
+        FP16_MATMUL_HEURISTIC_CONDITION = x.shape[0] * x.shape[1] >= 256
```

Comment:
I think we can make it a bit more general so that 1D tensors can also be supported?
```suggestion
        FP16_MATMUL_HEURISTIC_CONDITION = x.shape[:-1].numel() >= 256
```

Reply:
Yes, this is a bit more general although I’m not familiar enough with 1D shapes occurring in vLLM. Let me see if the suggestion works

Reply:
Fixed!